apps/aecore/lib/aecore/tx/data_tx.ex

```diff
@@ -194,12 +198,23 @@ defmodule Aecore.Tx.DataTx do
     end
   end
 
+  @spec chainstate_senders?(DataTx.t()) :: boolean()
+  def chainstate_senders?(%DataTx{type: type}) do
+    type.chainstate_senders?()
+  end
+
+  @spec senders_from_chainstate(DataTx.t(), Chainstate.t()) :: list(binary())
+  def senders_from_chainstate(%DataTx{payload: payload, type: type}, chainstate) do
+    type.senders_from_chainstate(payload, chainstate)
+  end
+
   @doc """
   Validates the transaction without considering state
   """
   @spec validate(DataTx.t(), non_neg_integer()) :: :ok | {:error, String.t()}
   def validate(
         %DataTx{fee: fee, type: type, senders: senders} = tx,
+        # FIXME
```

Comment:
Is this a placeholder for a future PR?

Reply:
`Chain.top_height()` shouldn't be used as block_height ever.

Reply:
I just opened an issue for that #626